fv3core/stencils/dyn_core.py

```diff
@@ -179,381 +197,415 @@ def dyncore_temporaries(shape):
-            if it != n_split - 1:
-                reqs_vector = comm.start_vector_halo_update(
-                    state.u_quantity, state.v_quantity, n_points=utils.halo
+            if not hydrostatic:
+                updatedzd.compute(
+                    state.dp_ref,
+                    state.zs,
+                    state.zh,
+                    state.crx,
+                    state.cry,
+                    state.xfx,
+                    state.yfx,
+                    state.wsd,
+                    dt,
                 )
-            else:
-                if spec.namelist.grid_type < 4:
-                    comm.synchronize_vector_interfaces(
-                        state.u_quantity, state.v_quantity
+
+                riem_solver3.compute(
+                    remap_step,
+                    dt,
+                    akap,
+                    state.cappa,
+                    state.ptop,
+                    state.zs,
+                    state.w,
+                    state.delz,
+                    state.q_con,
+                    state.delp,
+                    state.pt,
+                    state.zh,
+                    state.pe,
+                    state.pkc,
+                    state.pk3,
+                    state.pk,
+                    state.peln,
+                    state.wsd,
+                )
+
+                if self.do_halo_exchange:
+                    reqs["zh_quantity"] = self.comm.start_halo_update(
+                        state.zh_quantity, n_points=utils.halo
                     )
+                    if grid.npx == grid.npy:
+                        reqs["pkc_quantity"] = self.comm.start_halo_update(
+                            state.pkc_quantity, n_points=2
+                        )
+                    else:
+                        reqs["pkc_quantity"] = self.comm.start_halo_update(
+                            state.pkc_quantity, n_points=utils.halo
+                        )
+                if remap_step:
+                    pe_halo.compute(state.pe, state.delp, state.ptop)
+                if self.namelist.use_logp:
+                    raise Exception("unimplemented namelist option use_logp=True")
+                else:
+                    pk3_halo.compute(state.pk3, state.delp, state.ptop, akap)
+            if not hydrostatic:
+                if self.do_halo_exchange:
+                    reqs["zh_quantity"].wait()
+                    if grid.npx != grid.npy:
+                        reqs["pkc_quantity"].wait()
+                basic.multiply_constant(
+                    state.zh,
+                    state.gz,
+                    constants.GRAV,
+                    origin=(grid.is_ - 2, grid.js - 2, 0),
+                    domain=(grid.nic + 4, grid.njc + 4, grid.npz + 1),
+                )
+                if grid.npx == grid.npy and self.do_halo_exchange:
+                    reqs["pkc_quantity"].wait()
+                if self.namelist.beta != 0:
```

Comment:
Can you replicate or move the logic for the `NotImplementedError`s to the init?

Reply:
I can replicate it, though prefer not to move it for the use_logp one since the control flow logic being in place will make it easier to expand the options when we do so. the beta one can be moved easily because it has nothing to do with anything around it